test/unittest/StatisticsBackend/IsActiveTests.cpp

```diff
@@ -0,0 +1,474 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "fastdds/dds/domain/DomainParticipant.hpp"
+
+#include <database/database.hpp>
+#include <database/database_queue.hpp>
+#include <DatabaseUtils.hpp>
+#include <subscriber/StatisticsParticipantListener.hpp>
+
+using namespace eprosima::statistics_backend::subscriber;
+
+/**
+ * @brief Fixture for the is_active_tests
+ */
+class is_active_tests : public ::testing::Test
+{
+public:
+
+    void SetUp()
+    {
+        db = new DataBaseTest;
+        db->load_database(load_file(EMPTY_ENTITIES_DUMP_FILE));
+        host = db->hosts().begin()->second;
+        user = db->users().begin()->second;
+        process = db->processes().begin()->second;
+        domain = db->domains().begin()->second;
+        topic = db->topics().begin()->second.begin()->second;
+        participant = db->participants().begin()->second.begin()->second;
+        datawriter = db->get_dds_endpoints<DataWriter>().begin()->second.begin()->second;
+        datareader = db->get_dds_endpoints<DataReader>().begin()->second.begin()->second;
+        locator = db->locators().begin()->second;
```

Comment:
```suggestion
        locator = db->locators().begin()->second;
        StatisticsBackendTest::set_database(db);
```
Make clear that we are moving the ownership of the allocated db to the Backend, se there is no need to `delete` the database. Remove the `set_database` calls from the tests themselves, as is already in the setup.